### ISSUE TITLE
Add JSON-RPC method to get deterministic Snap-specific entropy

### ISSUE THREAD

**Issue by Mrtenz** [MetaMask/snaps-monorepo#860]
Right now for a Snap to get entropy, it could use `snap_getBip32Entropy` (or the BIP-44-equivalent), which derives a key from the user's secret recovery phrase. The downside of this is that other Snaps could simply take the same derivation path and get the same entropy. To solve this, we should introduce a new JSON-RPC method, `snap_getEntropy`, to get deterministic **Snap-specific** entropy.

As inputs we can use a combination of the user's secret recovery phrase, and the ID of the Snap. Since Snaps are never able to get the secret recovery phrase or root HD node, other Snaps will not be able to derive the same entropy.

We can use a similar approach to Ledger's password manager application, where it hashes the password (Snap ID in our case), splits the hash up in even chunks, and uses those as indices for the derivation path.

```ts
const MAGIC_PURPOSE = 0x...; // A random number we pick, should not be used for anything else.

const snapId = 'npm:foo';
const hash = keccak256(snapId); // Or other hashing algorithm.

const chunks: number[] = split(hash, 8); // Split hash into 8 chunks.
const { privateKey } = bip32Derive(secretRecoveryPhrase, [MAGIC_PURPOSE, ...chunks]);
```

The private key could then be used as entropy.

**Comment by Mrtenz** [comment]
The entropy could be used as seed to derive more keys from, so I don't think we need to include a coin type? We could have a salt if Snaps want to get different entropy for different purposes maybe.

```json
{
  "method": "snap_getEntropy",
  "params": {
    "salt": "foo"
  }
}
```

That could then be added to the snap ID before hashing.

**Comment by rekmarks** [comment]
I don't think we need a salt for v1 of this method; we should just assume that people know what to do with it (and we'll advertise what _not_ to do in our docs).

**Comment by Mrtenz** [comment]
Fixed in #940.

**Comment by Mrtenz** [comment]
Fixed in #940.
